impeller/renderer/context.h

```diff
@@ -19,31 +19,121 @@ class CommandBuffer;
 class PipelineLibrary;
 class Allocator;
 
+//------------------------------------------------------------------------------
+/// @brief      Do do anything rendering related with Impeller, you need a
+///             context.
+///
+///             Contexts are expensive to construct and typically you only need
+///             one in the process. The context represents a connection to a
+///             graphics or compute accelerator on the device.
+///
+///             If there are multiple context in a process, it would typically
+///             be for separation of concerns (say, use with multiple engines in
+///             Flutter), talking to multiple accelerators, or talking to the
+///             same accelerator using different client APIs (Metal, Vulkan,
+///             OpenGL ES, etc..).
+///
+///             Contexts are thread-safe. They may be created, used, and
```

Comment:
There is a small caveat here in that they may still need to be destructed on the thread they were created from, at least as long as they own their own concurrent worker loop.

Reply:
actually, I guess not on the thread it was constructed on - just not on a worker thread from an internal pool

Reply:
Added the clarification.